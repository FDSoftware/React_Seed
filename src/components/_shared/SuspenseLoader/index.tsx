// React
import React from "react";
// MUI Components
import { CircularProgress } from "@material-ui/core";

/**
 * ### Nombre del componente: `<SuspenseLoader>`
 * Componente loader que renderiza previo a las vistas.
 */

const SuspenseLoader = () => {
	return (
		<div
			className="loader"
			style={{
				position: "fixed",
				top: 0,
				left: 0,
				width: "100vw",
				height: "100vh",
				zIndex: 10000,
				backgroundColor: "#FFF",
				display: "flex",
				alignItems: "center",
				flexDirection: "column",
				justifyContent: "center",
				marginTop: "auto",
				marginBottom: "auto",
			}}
		>
			<CircularProgress
				size="6rem"
				className="loader__spinner"
				style={{
					color: "#e8eaef",
					margin: 0,
					padding: 0,
				}}
			/>
		</div>
	);
};
export default SuspenseLoader;
