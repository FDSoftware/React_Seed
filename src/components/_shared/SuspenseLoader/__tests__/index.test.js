/**
 * Tests SuspenseLoader Component
 *
 * @group unit
 */

// React
import React from "react";
// Testing Libs
import { render, screen } from "@testing-library/react";
// Component to Test
import SuspenseLoader from "components/_shared/SuspenseLoader";

describe("SuspenseLoader", () => {
	it("renders without crashing & matches snapshot", () => {
		const { asFragment } = render(<SuspenseLoader />);
		const suspenseLoader = screen.getByRole("progressbar");
		expect(suspenseLoader).toBeVisible();
		expect(asFragment()).toMatchSnapshot();
	});
});
