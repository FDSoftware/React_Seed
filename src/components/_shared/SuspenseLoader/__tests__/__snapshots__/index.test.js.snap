// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuspenseLoader renders without crashing & matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="loader"
    style="position: fixed; top: 0px; left: 0px; width: 100vw; height: 100vh; z-index: 10000; background-color: rgb(255, 255, 255); display: flex; align-items: center; flex-direction: column; justify-content: center; margin-top: auto; margin-bottom: auto;"
  >
    <div
      class="MuiCircularProgress-root loader__spinner MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style="width: 6rem; height: 6rem; color: rgb(232, 234, 239); margin: 0px; padding: 0px;"
    >
      <svg
        class="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
